### ISSUE TITLE
Running terminal process "lost" switching workspaces

### ISSUE THREAD

**Issue by ay0o** [Microsoft/vscode#64352]
- VSCode Version: 1.29.1
- OS Version: macOS 10.14.1

Steps to Reproduce:

1. Run VSCode and open a few terminals. Now, in one of them run for example a docker container (`docker run nginx`). Notice it's not running on background!! (`-d` is not specified)
2. Switch to a different workspace. 
3. Switch back to the previous workspace. First thing you can notice is that there is only one terminal, not all the instances that you have previously opened. Secondly, as there is only one new terminal, your running processes "disappeared" right? Actually they are still running but you can't just see them. For example, following the docker example, run `docker ps` and you will see the container is up and running until either you kill it or close vscode.

So, in summary, what's the issue? The Terminal status should be saved across workspaces, first to provide a consistent workspace experience, and secondly to prevent ghost processes.
